app/assets/javascripts/app/views/notifications_badge_view.js

```diff
@@ -0,0 +1,118 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+
+app.views.NotificationsBadge = app.views.Base.extend({
+  events:{
+    "click #notifications-badge": "toggleNotifDropdown"
+  },
+
+  initialize: function(){
+    $(document.body).click($.proxy(this.hideNotifDropdown, this));
+
+    this.notifications = [];
+    this.nextPage = 1;
+    this.perPage = 5;
+    this.hasMoreNotifs = true;
+    this.badge = this.$el;
+    this.dropdown = $('#notification_dropdown');
+    this.dropdownNotifications = this.dropdown.find(".notifications");
+    this.ajaxLoader = this.dropdown.find(".ajax_loader");
+  },
+
+  toggleNotifDropdown: function(evt){
```

Comment:
In this view there is just one dropdown so you could name this `toggleDropdown`. This applies to most functions in this view.